From 8bbda42706f1be82f88c95ef7fbc25fdc29a0775 Mon Sep 17 00:00:00 2001
From: Sagar Cheluvegowda <quic_scheluve@quicinc.com>
Date: Wed, 20 Aug 2025 10:55:20 -0700
Subject: [PATCH 10/16] PENDING: net: stmmac: Remove global variable for
 tracking the PHY interface

Refactor the code to eliminate the use of a global variable
for tracking the PHY interface and use the existing
phy_interface struct member instead.

Change-Id: I98f0f19f5ca1dda06014fcef8e761bb2f2fba849
Signed-off-by: Sagar Cheluvegowda <quic_scheluve@quicinc.com>
(cherry picked from commit 85c5559534441fce44ffc31e3c029cfa8c062be5)
---
 drivers/net/pcs/pcs-xpcs-qcom.c | 9 +++------
 1 file changed, 3 insertions(+), 6 deletions(-)

diff --git a/drivers/net/pcs/pcs-xpcs-qcom.c b/drivers/net/pcs/pcs-xpcs-qcom.c
index fab7f96cdbbf..1fd0cacc5601 100644
--- a/drivers/net/pcs/pcs-xpcs-qcom.c
+++ b/drivers/net/pcs/pcs-xpcs-qcom.c
@@ -2,7 +2,7 @@
 /*
  * Synopsys DesignWare XPCS platform device driver
  *
- * Copyright (c) 2024-2025 Qualcomm Innovation Center, Inc. All rights reserved.
+ * Copyright (c) Qualcomm Technologies, Inc. and/or its subsidiaries.
  */
 
 #include <linux/delay.h>
@@ -26,7 +26,6 @@
 #define XPCSERR(fmt, args...) \
 	pr_err(DRV_NAME " %s:%d " fmt, __func__, __LINE__, ## args)
 
-phy_interface_t g_interface;
 static const int xpcs_usxgmii_features[] = {
 	ETHTOOL_LINK_MODE_Pause_BIT,
 	ETHTOOL_LINK_MODE_Asym_Pause_BIT,
@@ -689,8 +688,6 @@ static int qcom_xpcs_select_mode(struct dw_xpcs_qcom *qxpcs, phy_interface_t int
 {
 	int ret;
 
-	g_interface = interface;
-
 	if (interface == PHY_INTERFACE_MODE_USXGMII ||
 	    interface == PHY_INTERFACE_MODE_10GBASER ||
 	    interface == PHY_INTERFACE_MODE_5GBASER) {
@@ -939,9 +936,9 @@ void qcom_xpcs_destroy(struct phylink_pcs *pcs)
 			continue;
 
 		qxpcs->id = entry;
-		compat = xpcs_find_compat(entry, g_interface);
+		compat = xpcs_find_compat(entry, qxpcs->phy_interface);
 		if (!compat) {
-			XPCSERR("Incompatible MII interface: %d\n", g_interface);
+			XPCSERR("Incompatible MII interface: %d\n", qxpcs->phy_interface);
 			ret = -ENODEV;
 			goto out;
 		}
